scripts/konflux/subscription-manager/subscription-manager-bro.sh

```diff
@@ -0,0 +1,311 @@
+SECRET_KONFLUX_WORKSPACE_PATH="/workspace/${SECRET_NAME_IN_KONFLUX}"
+SECRET_KEY="activation-key"
+SECRET_LOCAL_PATH="${SCRIPT_DIR}/${SECRET_KEY}"
+SECRET_INFO_URL='https://docs.engineering.redhat.com/pages/viewpage.action?pageId=407312060'
+
+RED_HAT_ORG_ID="11009103"
+TARGETS_LIST_FILE="/tmp/subscription-manager-bro-targets"
+
+# These were figured experimentally with the help of self-test subcommand.
+TARGET_BACKUP_PATHS=(
+    etc/pki/consumer
+    etc/pki/entitlement
+    etc/pki/product
+    etc/pki/product-default
+    etc/yum.repos.d
+    var/lib/rhsm
+    var/cache/ldconfig
+)
+
+
+function main {
+    if [[ "$#" == "0" ]] ; then
+        >&2 echo "Error: command is missing. See the usage below."
+        usage
+        exit 2
+    fi
+
+    local cmd="$1"
+    shift
+
+    local fn
+
+    case "$cmd" in
+    "help" | "--help" | "-h")
+        fn=usage ;;
+    "smuggle")
+        fn=smuggle ;;
+    "register")
+        fn=register ;;
+    "cleanup")
+        fn=cleanup ;;
+    "self-test")
+        fn=self_test ;;
+    "diff")
+        fn=assert_diff ;;
+    *)
+        >&2 echo "Error: unknown command '$1'; call '$SCRIPT_NAME help' to see the usage."
+        exit 2
+        ;;
+    esac
+
+    if [[ "$#" -gt "0" && "$cmd" != "diff" && "$cmd" != "register" ]]; then
+        >&2 echo "Error: too many arguments; call '$SCRIPT_NAME help' to see the usage."
+        exit 2
+    fi
+
+    "$fn" "$@"
+}
+
+function usage {
+    local example_target_dir="/mnt"
+
+    echo "Usage: $SCRIPT_NAME smuggle|register|cleanup|self-test"
+    echo
+    echo "This script enables access to RHEL RPMs during Konflux builds. The intended usage is as follows."
+    echo
+
+    echo -n "1. Make sure there is a '$SECRET_NAME_IN_KONFLUX' secret in Konflux with key name '$SECRET_KEY' and "
+    echo "the actual activation key as a value."
+    echo "   Find where to get the secret from ${SECRET_INFO_URL}"
+
+    echo -n "2. In a Tekton pipeline step before the container build, copy the subscription manager activation "
+    echo "key secret to the source workspace. Use:"
+    echo "   \$ <source-workspace>/$SCRIPT_NAME smuggle"
+    echo -n "   This expects the '$SECRET_NAME_IN_KONFLUX' secret to be mounted as a workspace with the same name "
+    echo "('$SECRET_KONFLUX_WORKSPACE_PATH')."
+
+    echo "3. Arrange Dockerfile stages to have UBI (normal) as an installer and other RHEL/UBI (any) as a target."
+    echo "   Make sure to match major versions: 8/8 is ok but 9/8 or 8/9 will result in errors."
+    echo "   Copy the target contents to some directory, e.g. ${example_target_dir}, in the installer stage."
+    echo "   See self-test Dockerfiles as examples."
+
+    echo "4. In the installer stage, register the container with the subscription manager. Use:"
+    echo "   \$ $SCRIPT_NAME register ${example_target_dir}"
+    echo -n "   It is possible to provide multiple target directories as arguments if the script is used to prepare "
```

Comment:
Yes, that will be `main` itself. We need to install some RHEL packages in rocksdb builder stage (yes, it will not live for long, but will be there through at least 4.4) and in the final stage.

I felt that declaring two installer stages, one for each of these, will look too heavy so added this feature of installing to multiple targets.